DataFormats/EcalRecHit/interface/EcalRecHitSoA.h

```diff
@@ -0,0 +1,16 @@
+#ifndef DataFormats_EcalRecHit_EcalRecHitSoA_h
+#define DataFormats_EcalRecHit_EcalRecHitSoA_h
+
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+
+GENERATE_SOA_LAYOUT(EcalRecHitSoALayout,
+                    SOA_SCALAR(uint32_t, size),
+                    SOA_COLUMN(uint32_t, id),
```

Comment:
If this is the detId, I think it would be better to rename to `detId`. If it is not, then for PF we need the detId propagated.

Reply:
It's okay then I can use as is.